### PR TITLE
support different Pinecone initializations depending on the version

### PR DESCRIPTION
**Pull request by DosticJelena**
# Description

Adding support for different initialization implementations depending on the Pinecone client version.

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested by installing different versions of the Pinecone client and trying to create an index (both regular and serverless for the new version) from documents, and using the `PineconeVectorStore` to query the index. The process ran smoothly and as expected in both cases.

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama_index/vector_stores/pinecone.py [hunk 1]
```diff
@@ -10,6 +10,9 @@
 from functools import partial
 from typing import Any, Callable, Dict, List, Optional, cast
 
+from packaging import version
+from pkg_resources import get_distribution
```
Comment (logan-markewich) [inline llama_index/vector_stores/pinecone.py:14]: isn't `pkg_resources` deprecated? I've been meaning to remove an use of it from our codebase

Comment (aulorbe) [reply llama_index/vector_stores/pinecone.py:14]: (Of note, though, is that LlamaIndex is still on `3.8`, so I don't know how much this rly matters.)

Comment (logan-markewich) [reply llama_index/vector_stores/pinecone.py:14]: @DosticJelena actually, we support 3.8 to 3.12 technically

Comment (logan-markewich) [reply llama_index/vector_stores/pinecone.py:14]: Actual python version is [here](https://github.com/run-llama/llama_index/blob/17e280b2bade6b5d3e3b5af6c9502f572066c201/pyproject.toml#L60)

**Comment by aulorbe** [comment]
@DosticJelena Can we close this?

**Pull request closed without merging**